### ISSUE TITLE
Formatting issues with non closing html tags

### ISSUE THREAD

**Issue by m1212e** [microsoft/vscode#112182]
[Coming from here](https://github.com/OmniSharp/omnisharp-vscode/issues/4284) I'll quickly want to confirm if this is intended behavior. If it is, why? It doesn't really make a lot of sense in my eyes.

**Comment by aeschli** [comment]
I tried your example
- all extra extensions disbaled
- create a new file 'foo.html'. Verify that the editor's language mode is `HTML`

```
<html>

<body>
	                <div>
		<textarea />
	    </div>

</body>


</html>
```
- run format

```
<html>

<body>
	<div>
		<textarea />
	</div>

</body>


</html>
```

I don't see any strange indentation,

Can you add your sample?

**Comment by m1212e** [comment]
I didn't notice it myself in pure HTML either, but as you saw have been told that it's caused by the vscode formatter. The code I've got is in a .razor file and therefore contains things not given in pure html, like @if statements which maybe could break the formatter. Nonetheless, here you go:
This:

```
        <div class="col p-0">
            <EditForm Model="Paragraph" class="h-100">
                <textarea value="@Paragraph.Text" @oninput="@OnInputChange" disabled="@this.Paragraph.IsLocked"
                    placeholder="Text..." class="form-control h-100 custom-input-field px-0 p-2"></textarea>
            </EditForm>
        </div>
```

And this:

```
    <div class="card my-0 col @(commentsExpanded ? "comments-expanded ml-2" : "comments") p-0">
        <textarea placeholder="Kommentare..."
            class="form-control h-100 custom-input-field px-0 p-2 no-scrollbar" />
    </div>
```

**Comment by aeschli** [comment]
The HTML formatter requires HTML or something close to it. Just using it with Razor syntax will have issues, that's to be expected. Maybe the Razor extension turns the Razor code to HTML, runs the formatter on that and applies the changes back for Razor.
I think the Razor extension team should analyses this more closer first. If they use the HTML formatter with proper HTML and it doesn't work well, then it would be great if they can file an issue with the HTML sample code.

**Comment by ajaybhargavb** [comment]
@aeschli I disabled all extensions and tried it out and I am still seeing the strange indentation.
![htmlformatter](https://user-images.githubusercontent.com/1579269/101819783-90b51480-3ada-11eb-84a0-fc5d702ef407.gif)

Here is the version I used,
![image](https://user-images.githubusercontent.com/1579269/101819877-ab878900-3ada-11eb-94af-a12bda081d92.png)

Is there maybe a setting that drives this behavior?

**Comment by aeschli** [comment]
Ok, looks like this is fixed in 1.52.
